serial_protocol.md

```diff
@@ -0,0 +1,78 @@
+#Protocol:
+During the first use, or when the chiplet changes environments a “NETWORK” call is expected to initialize the wifi parameters.
+
+Every time a serial connection is established we will expect a “INIT” call after which any subsequent calls can be made, until the connection is closed. 
+
+In the following examples we use $ to represent variables. For example $Host represents your firebase host.
+
+##Response Format Byte
+All responses will be prefixed with one of the following bytes signifying the response type.
+```
+  + If response is ok and a raw string value.
+  * If response is ok and a raw string value prefixed by count of bytes in response then new line.
+  # If response is ok and a integer value.
+  . If response is ok and a float value.
```

Comment:
json doesn't but I think there is a lot of value in providing the distinction for c++ parsing. So you can do a itoa() call with confidence knowing that the result will parse. I think we can determine pretty easily in the library (check for a '.').